lib/collection/src/collection/resharding.rs

```diff
@@ -214,8 +214,49 @@ impl Collection {
             },
         }
 
+        let is_in_progress = match self.resharding_state().await {
+            Some(state) if state.matches(&resharding_key) => {
+                if !force && state.stage >= ReshardStage::ReadHashRingCommitted {
+                    return Err(CollectionError::bad_request(format!(
+                        "can't abort resharding {resharding_key}, \
+                         because read hash ring has been committed already, \
+                         resharding must be completed",
+                    )));
+                }
+
+                true
+            }
+
+            Some(state) => {
+                log::warn!(
+                    "aborting resharding {resharding_key}, \
+                     but another resharding is in progress:\n\
+                     {state:#?}"
+                );
+
+                false
+            }
+
+            None => {
+                log::warn!(
+                    "aborting resharding {resharding_key}, \
+                     but resharding is not in progress"
+                );
+
+                false
+            }
+        };
+
+        // We need to abort all resharding transfers
+        let resharding_transfers = shard_holder
+            .get_transfers(|t| t.method == Some(ShardTransferMethod::ReshardingStreamRecords));
+        for transfer in resharding_transfers {
+            self.abort_shard_transfer(transfer.key(), Some(&shard_holder))
+                .await?;
+        }
```

Comment:
No, only the transfers that belong to our resharding key.

I also vote to do it inside the shard holder if possible, not here.